### PR TITLE
Introduce encrypted scramble set PDFs, and reorganize `.zip` structure.

### PR DESCRIPTION
**Pull request by lgarron**
New top-level folder structure:
- `computer-display`: One PDF for each group, encrypted using the same password as the main `.zip` file.
- `printing`: All the old unencrypted PDFs.
- `interchange`: All the old non-PDF formats.

---------------

I realize there's some code redundancy in this PR, but I've tried to make this PR as clearly "correct" as possible without fancy tricks or refactoring.

It's also not a full solution to #382, but gets us a significant part of the way there.

Possible improvements:
- Encrypt each PDF using a separate passcode, and give the Delegate a file with the passcodes so that they can hand out decryption powers more granularly.
- Also include a .zip of the computer-display folder, for easy distribution. (Sharing a folder of files without re-zipping is not simple.)

### CODE REVIEW

**Comment by lgarron** [comment]
Sample `.zip` file:
[Test Competition.zip](https://github.com/thewca/tnoodle/files/2632977/Test.Competition.zip)

**Comment by lgarron** [comment]
Sample `.zip` with individual passcodes per scramble set PDF: 
[Test Competition.zip](https://github.com/thewca/tnoodle/files/2633126/Test.Competition.zip)

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 1]
```diff
@@ -357,8 +370,7 @@ private static PdfReader createPdf(String globalTitle, Date creationDate, Scramb
         doc.close();
 
         // TODO - is there a better way to convert from a PdfWriter to a PdfReader?
```
[the remark this answers is not in the document]
Comment (campos20) [reply webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: I agree we should split createPdf. I was also considering a different class for fmc and we can call it with addFmc(). Fmc sheet is way different from regylar scrambles. It looks messy to code together.

**Comment by lgarron** [comment]
@jfly, could you review again?

Here's the latest structure:
screen shot 2018-12-02 at 01 26 44

**Comment by lgarron** [comment]
`.zip`: [Scrambles for Test Competition.zip](https://github.com/thewca/tnoodle/files/2636817/Scrambles.for.Test.Competition.zip)

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 2]
```diff
@@ -276,21 +281,26 @@ private static PdfReader createPdf(String globalTitle, Date creationDate, Scramb
                 attemptRequest.event = "333bf";
                 attemptRequest.colorScheme = scrambleRequest.colorScheme;
 
-                PdfReader pdfReader = createPdf(globalTitle, creationDate, attemptRequest, locale);
+                PdfReader pdfReader = new PdfReader(createPdf(globalTitle, creationDate, attemptRequest, locale, null).toByteArray());
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Could you put a quick comment here about why password is null?

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 3]
```diff
@@ -1312,12 +1338,23 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
         ZipOutputStream zipOut = new ZipOutputStream(baosZip);
         HashMap<String, Boolean> seenTitles = new HashMap<String, Boolean>();
 
+        // Computer display zip
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Could you add a quick comment here about the goals of this zip file? I think it's easy to get confused here about if zip files are created just to easily hold files vs to encrypt stuff. I'm thinking something like:

> This zip file is intentionally *not* password protected, it's just an easy way to distribute a collection of files. Any files we put in this zip must be encrypted themselves.

(feel free to reword, but I hope the gist of what I'm trying to say makes sense)

**Comment by lgarron** [comment]
This change has been approved by the Board for a TNoodle release!

**Comment by jfly** [comment]
LGTM!

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java (hunk 2 above)
Comment (lgarron) [reply webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Done!

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java (hunk 3 above)
Comment (lgarron) [reply webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Done, thanks!

**Pull request merged**